### PR TITLE
[tesla] Add null annotations

### PR DESCRIPTION
**Pull request by lsiepel**
- Fix null annotations
- Fix compile warnings
- Updated and moved DTO's to dto folder
 
Only form code perspective, needs some testing to be sure no regression occures.

Test jar 4.3.0 (should also work with 4.2.x) https://1drv.ms/u/s!AnMcxmvEeupwj4oOcmeYDamnLAcyAg?e=k2nzcw

### CODE REVIEW

**Comment by lsiepel** [comment]
@kaikreuzer would you be able to test this binding? As you seem one of the few that not be forced to use the fleet api yet.

**Comment by kaikreuzer** [comment]
Hey @lsiepel! Thanks for the clean-ups! I've installed this version on my production system and will watch it over the day tomorrow. So far, it seems to work and I do not see any issues in my logs. Will keep you posted.

**Comment by kaikreuzer** [comment]
Unfortunately, I do not seem to receive any updates anymore - so I would say that this PR seems to break the functionality.
I'd need more time to deeply look into it in order to debug the issue...

**Comment by lsiepel** [comment]
> ook into it in order to debug the issue...

Too bad, looking at the code again i can't spot the obvious error. If you can provide debug logs and your observations i can try to look at it. But maybe it is simpler for you to just debug it when having the car connected.

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaEventEndpoint.java [hunk 1]
```diff
@@ -54,25 +54,20 @@ public class TeslaEventEndpoint implements WebSocketListener, WebSocketPingPongL
...
-            this.endpointId = "TeslaEventEndpoint-" + uid.getAsString();
+        this.endpointId = "TeslaEventEndpoint-" + uid.getAsString();
 
-            String name = ThingWebClientUtil.buildWebClientConsumerName(uid, null);
-            client = webSocketFactory.createWebSocketClient(name);
-            this.client.setConnectTimeout(TIMEOUT_MILLISECONDS);
-            this.client.setMaxIdleTimeout(IDLE_TIMEOUT_MILLISECONDS);
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
```
Comment (lsiepel) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaEventEndpoint.java:75]: @kaikreuzer  without a debug log it is hard to find the culprit. These lines might be related. I would not expect a runtimeexception here so i removed it. Maybe i shouldn't have done that.

Any way a log would be highly appreciated.

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/protocol/dto/VehicleData.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @author Kai Kreuzer - Initial contribution
+ */
+public class VehicleData {
+
+    @SerializedName("id")
+    public Long id;
+    @SerializedName("user_id")
+    public int userId;
+    @SerializedName("vehicle_id")
+    public int vehicleId;
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/protocol/dto/VehicleData.java:32]: This used to be a String and `int` is causing parsing exceptions. I assume this was changed by mistake?
```suggestion
    public String vehicleId;
```

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Thanks @lsiepel, I finally got it running (my local development env was broken due to https://github.com/openhab/openhab-distro/pull/1697, which I only figured out today.

It seems that your code works well after all - there is just one small bug that causes it to break. Once this is fixed, it looks good!

**Comment by lsiepel** [comment]
Done, I leave it to you if you want to merge it for 4.3.0 or not.

**Review by kaikreuzer — APPROVED** [review]
Let's merge, thanks!

**Pull request merged**